### ISSUE TITLE
Remove DFParser

### ISSUE THREAD

**Issue by tustvold** [apache/datafusion#4808]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

DataFusion contains a custom `DFParser` that extends `sqlparser` and overrides its parsing of `DESCRIBE TABLES` and `CREATE EXTERNAL TABLE`.

**Describe the solution you'd like**

Looking at sqlparser the only functionality missing upstream is understanding the `COMPRESSION` keyword in `CREATE EXTERNAL TABLE`. This feels like a very simple upstream contribution that would then allow removing `DFParser` and its corresponding `Statement` abstraction.

**Describe alternatives you've considered**

We could not do this

**Additional context**

FYI @andygrove @alamb  @liukun4515

**Comment by tustvold** [comment]
Actually I think I would like to use Statement, to remove the DDL from LogicalPlan

**Comment by unconsolable** [comment]
There are some differences in datafusion's `CreateExternalTable` and sqlparser's `CreateTable`, for example `has_header` , `table_partition_cols` and `file_compression_type` are not exist in sqlparser's `CreateTable`. Are there any way to create custom `Statement` in `sqlparser`?

**Comment by tustvold** [comment]
The partitoning is under one of the Hive options, not sure about the others. I believe there is support for custom dialects, but in this case I'd vote for upstreaming anything missing - we don't have any especially esoteric needs AFAIK

**Comment by Jefffrey** [comment]
I'd like to take this on. I'll try do the migration in steps to make PRs more digestible and ensure no breakages/regressions.

First step is here: #8703

**Comment by alamb** [comment]
Related conversation in https://github.com/sqlparser-rs/sqlparser-rs/issues/1080

From what I know, the `CREATE EXTERNAL TABLE` syntax is not based on any other dialect and thus is DataFusion specific. I think it may not belong upstream in sqlparser-rs 

The COPY syntax I tried to base on DuckDB's copy syntax (which is different than Postgres), so if we added DuckDB style COPY support to sqlparser-rs we could probably migrate DataFusion to use it

**Comment by alamb** [comment]
I don't think we plan to do this / remove the EXTERNAL TABLE syntax so let's close this ticket to avoid confusion
